### ISSUE TITLE
TypeError: type object got multiple values for keyword argument 'virtual_host'

### ISSUE THREAD

**Issue by chotee** [pretix/pretix#358]
When running pretix from a development server,

```$ python manage.py runserver 192.168.56.103:8080```

Certain urls (like cart/add) give the following error:

```
[14/Dec/2016 14:26:37] "GET /marijn/sha/ HTTP/1.1" 200 6404
[14/Dec/2016 14:26:37] "GET /static/CACHE/css/main.7ee474e5da50.css HTTP/1.1" 200 362314
ERROR 2016-12-14 14:26:43,319 django.request exception Internal Server Error: /marijn/sha/cart/add
Traceback (most recent call last):
  File "/var/pretix/venv/lib/python3.4/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/var/pretix/venv/lib/python3.4/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/var/pretix/venv/lib/python3.4/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/pretix/venv/lib/python3.4/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/var/pretix/venv/lib/python3.4/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/pretix/src/pretix/presale/views/cart.py", line 148, in post
    return self.do(self.request.event.id, items, self.request.session.session_key)
  File "/opt/pretix/src/pretix/base/views/async.py", line 25, in do
    res = self.task.apply_async(args=args)
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/task.py", line 565, in apply_async
    **dict(self._get_exec_options(), **options)
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 348, in send_task
    with self.producer_or_acquire(producer) as P:
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 403, in producer_or_acquire
    producer, self.amqp.producer_pool.acquire, block=True,
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/amqp.py", line 502, in producer_pool
    self.app.pool,
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 609, in pool
    self._pool = self.connection().Pool(limit=limit)
  File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 386, in connection
    'BROKER_CONNECTION_TIMEOUT', connect_timeout
  File "/var/pretix/venv/lib/python3.4/site-packages/kombu/connection.py", line 167, in __init__
    params.update(parse_url(hostname))
  File "/var/pretix/venv/lib/python3.4/site-packages/kombu/utils/url.py", line 35, in parse_url
    password=password, virtual_host=path, **query)
TypeError: type object got multiple values for keyword argument 'virtual_host'
Internal Server Error: /marijn/sha/cart/add
```

**Comment by raphaelm** [comment]
Can you show me the ``[celery]`` part of your ``pretix.cfg``, if there is any?

**Comment by chotee** [comment]
In_ my /etc/pretix/pretix.cfg I have the following:
[celery]
backend=redis://127.0.0.1?virtual_host=1
broker=redis://127.0.0.1?virtual_host=2

I redid the the dev installation into a new python "env"  following "https://docs.pretix.eu/en/latest/development/setup.html#your-local-python-environment" But I keep hitting this issue. This machine has a non-dev installation of pretix already. I'll retry with a clean machine where I only try to install the dev-environment.

**Comment by chotee** [comment]
Okay... just did a clean reinstall, now properly following the "The development setup" page on a clean Debian 8.6 Debian machine and now it does work (for now).  

This issue is probably caused by me trying to be smarter then I am and skrewing up the install in weird and wonderful ways.

**Comment by eomot** [comment]
I get the same error when installing like described here: 
https://docs.pretix.eu/en/latest/admin/installation/manual_smallscale.html
So basically with pip.

Happens when starting the service pretix-worker or adding a product to the cart.
System is debian 8.5
Any ideas ?

The relevant part of my pretix.cfg:

[redis]
location=redis://127.0.0.1/0
sessions=true

[celery]
backend=redis://127.0.0.1?virtual_host=1
broker=redis://127.0.0.1?virtual_host=2

And the error message:

`systemd[1]: Started pretix background worker.
celery[27951]: Traceback (most recent call last):
celery[27951]: File "/var/pretix/venv/bin/celery", line 11, in <module>
celery[27951]: sys.exit(main())
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/__main__.py", line 14, in main
celery[27951]: _main()
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/celery.py", line 326, in main
celery[27951]: cmd.execute_from_commandline(argv)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/celery.py", line 488, in execute_from_co
celery[27951]: super(CeleryCommand, self).execute_from_commandline(argv)))
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/base.py", line 281, in execute_from_comm
celery[27951]: return self.handle_argv(self.prog_name, argv[1:])
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/celery.py", line 480, in handle_argv
celery[27951]: return self.execute(command, argv)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/celery.py", line 412, in execute
celery[27951]: ).run_from_argv(self.prog_name, argv[1:], command=argv[0])
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/worker.py", line 221, in run_from_argv
celery[27951]: return self(*args, **options)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/base.py", line 244, in __call__
celery[27951]: ret = self.run(*args, **kwargs)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/bin/worker.py", line 255, in run
celery[27951]: **kwargs)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/worker/worker.py", line 99, in __init__
celery[27951]: self.setup_instance(**self.prepare_args(**kwargs))
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/worker/worker.py", line 120, in setup_instan
celery[27951]: self._conninfo = self.app.connection_for_read()
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 752, in connection_for_re
celery[27951]: return self._connection(url or self.conf.broker_read_url, **kwargs)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/celery/app/base.py", line 828, in _connection
celery[27951]: 'broker_connection_timeout', connect_timeout
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/kombu/connection.py", line 181, in __init__
celery[27951]: url_params = parse_url(hostname)
celery[27951]: File "/var/pretix/venv/lib/python3.4/site-packages/kombu/utils/url.py", line 35, in parse_url
celery[27951]: return dict(transport=scheme, hostname=host, port=port, userid=user, password=password, virtual_host=p
celery[27951]: TypeError: type object got multiple values for keyword argument 'virtual_host'`

**Comment by raphaelm** [comment]
Can you show me the output of ``pip freeze`` to check the dependency versions? :)

**Comment by eomot** [comment]
amqp==2.1.4
beautifulsoup4==4.5.3
billiard==3.5.0.2
bleach==1.5.0
celery==4.0.2
cffi==1.9.1
chardet==2.3.0
contextlib2==0.5.4
cryptography==1.7.1
csscompressor==0.9.4
cssutils==1.0.1
dj-static==0.0.6
Django==1.10.5
django-appconf==1.0.2
django-bootstrap3==7.1.0
django-compressor==2.1
django-formset-js-improved==0.5.0.1
django-formtools==1.0
django-jquery-js==3.1.1
django-libsass==0.7
django-markup==1.1
django-otp==0.3.8
django-redis==4.1.0
django-statici18n==1.2.1
easy-thumbnails==2.3
gunicorn==19.6.0
html5lib==0.9999999
idna==2.2
inlinestyler==0.2.3
kombu==4.0.2
libsass==0.12.3
lxml==3.7.2
Markdown==2.6.7
msgpack-python==0.4.8
mt-940==3.2
mysqlclient==1.3.9
olefile==0.44
paypalrestsdk==1.12.0
Pillow==4.0.0
ply==3.9
pretix==1.0.0b2
pyasn1==0.1.9
pycparser==2.13
pyOpenSSL==16.2.0
PyPDF2==1.26.0
python-dateutil==2.4.2
python-u2flib-server==4.0.1
pytz==2016.10
PyYAML==3.12
raven==5.32.0
rcssmin==1.0.6
redis==2.10.5
reportlab==3.2.0
requests==2.12.4
rjsmin==1.0.12
six==1.10.0
slimit==0.8.1
static3==0.6.1
stripe==1.22.3
typing==3.5.3.0
vine==1.1.3

**Comment by raphaelm** [comment]
Weird. I did not experience this before or know what it could be right now :(

**Comment by raphaelm** [comment]
Ah, can you try
```
backend=redis://127.0.0.1/1
broker=redis://127.0.0.1/2
```
instead of
```
backend=redis://127.0.0.1?virtual_host=1
broker=redis://127.0.0.1?virtual_host=2
```

**Comment by raphaelm** [comment]
Maybe this is just a documentation bug

**Comment by eomot** [comment]
yep, this fixes the problem. thank you!
it may not be the right right thread to discuss this (sorry), but in the documentation there are two more "problems".
1. I also needed to pip3 install typing
2. I needed to python -m pretix migrate first and then rebuild, since there was an error when trying to rebuild first, since the mysql database is still empty at that point. So maybe just flip the order in the documentation?

Thank you for your work and a great project!
